src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -174,17 +210,30 @@ private void UpdateStateMachine()
             }
 
             _workspace.Options = options;
-
             if (options.GetOption(KeybindingResetOptions.NeedsReset))
             {
                 ShowGoldBar();
             }
+
+            _isFirstRun = false;
+
+            bool StatusChanged(ReSharperStatus current, ReSharperStatus last)
+            {
+                // if ReSharper was suspended and user closed and restarted Visual Studio, 
+                // consider that a status change so we reset the gold bar
+                if (current == ReSharperStatus.Suspended && last == ReSharperStatus.Suspended && _isFirstRun)
```

Comment:
I would just say, if (_isFirstRun) return true since this "StatusChanged" is just perf optimization.